sdk/v2/authz/role_assignments.go

```diff
@@ -0,0 +1,132 @@
+package authz
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	rm "github.com/brigadecore/brigade/sdk/v2/internal/restmachinery"
+	libAuthz "github.com/brigadecore/brigade/sdk/v2/lib/authz"
+	"github.com/brigadecore/brigade/sdk/v2/meta"
+	"github.com/brigadecore/brigade/sdk/v2/restmachinery"
+)
+
+// PrincipalType is a type whose values can be used to disambiguate one type of
+// principal from another. For instance, when assigning a Role to a principal
+// via a RoleAssignment, a PrincipalType field is used to indicate whether the
+// value of the PrincipalID field reflects a User ID or a ServiceAccount ID.
+type PrincipalType string
+
+const (
+	// PrincipalTypeServiceAccount represents a principal that is a
+	// ServiceAccount.
+	PrincipalTypeServiceAccount PrincipalType = "SERVICE_ACCOUNT"
+	// PrincipalTypeUser represents a principal that is a User.
+	PrincipalTypeUser PrincipalType = "USER"
+)
+
+// PrincipalReference is a reference to any sort of security principal (human
+// user, service account, etc.)
+type PrincipalReference struct {
+	// Type qualifies what kind of principal is referenced by the ID field-- for
+	// instance, a User or a ServiceAccount.
+	Type PrincipalType `json:"type,omitempty"`
+	// ID references a principal. The Type qualifies what type of principal that
```

Comment:
There's no question that if you're looking directly at this code, the repeated description of the `Type` field in the description of the `ID` field seems redundant.

Since the SDK is intended for broad consumption, consider someone who isn't looking directly at this code or even the godocs generated from it, but is instead only seeing dynamic context help in their IDE, as in the attached screen cap. I think the repeated description of the `Type` field provides critical context that such a user needs to know about.

<img width="1017" alt="Screen Shot 2021-01-27 at 12 51 22 PM" src="https://user-images.githubusercontent.com/1821014/106032394-6aedca00-609e-11eb-8cf3-88431752aa36.png">